lib/class-wp-theme-json-gutenberg.php

```diff
@@ -863,6 +871,15 @@ protected static function get_blocks_metadata() {
 				}
 				static::$blocks_metadata[ $block_name ]['elements'][ $el_name ] = implode( ',', $element_selector );
 			}
+
+			// If the block has style variations, append their selectors to the block metadata.
+			if ( ! empty( $block_type->styles ) ) {
+				$style_selectors = array();
+				foreach ( $block_type->styles as $style ) {
+					$style_selectors[ $style['name'] ] = static::append_to_selector( '.is-style-' . $style['name'] . '.is-style-' . $style['name'], static::$blocks_metadata[ $block_name ]['selector'] );
```

Comment:
Just double checking — this adds the classname twice, which increases specificity. Is this to ensure that styles included in `theme.json` override the defaults for the style variation provided in CSS stylesheets? If so, that sounds like a decent way to ensure it overrides them to me! Is it worth adding a comment prior to this line explaining the extra specificity?

Reply:
Yeah, I duplicated the classname to make sure these styles override the block core styles. To answer your question above, I didn't do the same in the editor because`editor-styles-wrapper` already guarantees the needed specificity. It's a good idea to add a comment here though!